opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchQueryRequest.java

```diff
@@ -189,6 +189,9 @@ public OpenSearchResponse searchWithPIT(Function<SearchRequest, SearchResponse>
       // Set sort field for search_after
       if (this.sourceBuilder.sorts() == null) {
         this.sourceBuilder.sort(DOC_FIELD_NAME, ASC);
+        // Workaround to preserve sort location more exactly,
```

Comment:
if this is workaround, could u add the long-term solution issue?

Reply:
#3064